shopinvader_schema_invoice/schemas/invoice.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Camptocamp SA
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from extendable_pydantic import StrictExtendableBaseModel
+
+from .amount import InvoiceAmount
+
+
+class Invoice(StrictExtendableBaseModel, extra="ignore"):
+    id: int
+    name: str
+    date_invoice: date
+    date_due: date | None = None
+    state: str
+    payment_state: str
+    ref: str | str
+    payment_reference: str | str
+    amount: InvoiceAmount | None
+
+    @classmethod
+    def from_one(cls, odoo_rec):
```

Comment:
from_... what?

Reply:
Here too I prefer `from_invoice`. Or `from_account_move`.
I find it more explicit and readable.

If the ORM had types it would be `def from_invoice(cls, record: AccountMove) -> Self`.